### PR TITLE
Add attributes_present, attributes_absent.

### PR DESCRIPTION
**Pull request by jechol**
Closes https://github.com/ash-project/ash/issues/1015

# Contributor checklist

- [ ] Bug fixes include regression tests
- [ ] Chores
- [ ] Documentation changes
- [X] Features include unit/acceptance tests
- [ ] Refactoring
- [ ] Update dependencies

### CODE REVIEW

File: lib/ash/resource/validation/builtins.ex [hunk 1]
```diff
@@ -197,6 +197,27 @@ defmodule Ash.Resource.Validation.Builtins do
...
+
+  If no options are provided, validates that they are all present.
+
+  ## Options
+
+  #{Spark.Options.docs(Ash.Resource.Validation.AttributePresent.opt_schema())}
+  """
+  @spec attribute_present(attributes :: atom | list(atom), opts :: Keyword.t()) ::
+          Validation.ref()
+  def attribute_present(attributes, opts \\ []) do
```
Comment (zachdaniel) [inline lib/ash/resource/validation/builtins.ex:211]: Maybe we should use the plural `attributes_present` here? i.e `validate attribute_present([:foo, :bar], at_least: 1, at_most: 1)` vs `validate attribute_present([:foo, :bar], at_least: 1, at_most: 1)`, the former looks better.

**Review by zachdaniel — DISMISSED** [review]
Just some naming discussions

**Comment by jechol** [comment]
@zachdaniel pluralized now.

**Comment by jechol** [comment]
@zachdaniel conflict fixed now.

**Comment by zachdaniel** [comment]
🚀 Thank you for your contribution! 🚀

**Pull request merged**